yellowbrick/features/pcoords.py

```diff
@@ -333,34 +346,15 @@ def fit(self, X, y=None, **kwargs):
         self : instance
             Returns the instance of the transformer/visualizer
         """
+        # Determine the features, classes, and colors
+        super(ParallelCoordinates, self).fit(X, y)
 
         # Convert from pandas data types
         if is_dataframe(X):
-            # Get column names before reverting to an np.ndarray
-            if self.features_ is None:
-                self.features_ = np.array(X.columns)
-
             X = X.values
         if is_series(y):
             y = y.values
 
-        # Assign integer labels to the feature columns from the input
-        if self.features_ is None:
-            self.features_ = np.arange(0, X.shape[1])
-
-        # Ensure that all classes are represented in the color mapping (before sample)
-        # NOTE: np.unique also specifies the ordering of the classes
-        if self.classes_ is None:
-            self.classes_ = [str(label) for label in np.unique(y)]
-
-        # Create the color mapping for each class
-        # TODO: Allow both colormap, listed colors, and palette definition
-        # TODO: Make this an independent function or property for override!
-        color_values = resolve_colors(
-            n_colors=len(self.classes_), colormap=self.colormap, colors=self.color
-        )
-        self._colors = dict(zip(self.classes_, color_values))
-
```

Comment:
A lot of duplicate code was removed and functionality unified in the super class - making things easier to test and change!

Reply:
So slick!